openedx_learning/apps/authoring/components/api.py

```diff
@@ -191,6 +192,20 @@ def create_next_component_version(
             # content represented by our key from the next version. Otherwise,
             # we add our key->content_pk mapping to the next version.
             if content_pk is not None:
+                if isinstance(content_pk, bytes):
+                    file_path, file_content = key, content_pk
+                    media_type_str, _encoding = mimetypes.guess_type(file_path)
+                    # We use "application/octet-stream" as a generic fallback media type, per
+                    # RFC 2046: https://datatracker.ietf.org/doc/html/rfc2046
+                    media_type_str = media_type_str or "application/octet-stream"
+                    media_type = contents_api.get_or_create_media_type(media_type_str)
+                    content = contents_api.get_or_create_file_content(
+                        component.publishable_entity.learning_package.id,
```

Comment:
```suggestion
                        component.learning_package_id,
```

This should remove two unnecessary model lookups.